Tella/Utils/Extensions/PHAssetExtension.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  PHAssetExtension.swift
+//  Tella
+//
+//  Created by Dhekra Rouatbi on 17/6/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Photos
+
+extension PHAsset {
+    
+    /// An extension on `PHAsset` to asynchronously fetch image data.
+    ///
+    /// This function uses Swift's concurrency features to provide an async/await interface for fetching image data
+    /// from a `PHAsset` object.
+    ///
+    /// - Returns: The image data as a `Data` object, or nil if no data is available.
+    /// - Throws: An error if the image data could not be fetched.
+    ///
+    
+    func getDataFromAsset() async throws -> Data? {
+        
+        return try await withCheckedThrowingContinuation { continuation in
+            
+            PHImageManager.default().requestImageDataAndOrientation(for: self, options: nil) { (data, uti, orientation, info) in
+                
+                guard let data else {
+                    guard let error = info?[PHImageErrorKey] as? Error else {
+                        continuation.resume(throwing: RuntimeError("Unknown error occurred"))
```

Comment:
what do you think of changing the nested guard statement for a `if let`? I think it could be cleaner this way:

```
if let data = data {
    continuation.resume(returning: data)
} else if let error = info?[PHImageErrorKey] as? Error {
    continuation.resume(throwing: error)
} else {
    continuation.resume(throwing: PHAssetError.unknownError)
}
```

Reply:
We could apply the same in the `getAVAssetUrl()` if we decide to go that way

Reply:
I did that but then I changed with a `guard let` 🤔